src/file_response.cpp

```diff
@@ -31,7 +31,11 @@ namespace httpserver {
 
 MHD_Response* file_response::get_raw_response() {
     int fd = open(filename.c_str(), O_RDONLY);
-    size_t size = lseek(fd, 0, SEEK_END);
+    if (fd == -1) return nullptr;
+
+    off_t size = lseek(fd, 0, SEEK_END);
```

Comment:
Type of `size` taking return value of `lseek()` should be `off_t`. That was introduced wrong with f9b76910. Could be fixed in a separate commit or just fixed here en passant.